recipe/meta.yaml

```diff
@@ -1,17 +1,17 @@
-{% set name = "re2" %}
-{% set version = "2023-03-01" %}
+{% set version = "2023-06-02" %}
 {% set dotversion = version.replace('-', '.') %}
-# switching from C++14 to C++17 unfortunately introduces a different ABI, see #61;
-# create a fake version bump so we can migrate the new version safely
-{% set abi_bump_version = dotversion.replace('.01', '.02') %}
+
+# see https://github.com/google/re2/blob/2023-06-01/CMakeLists.txt#L32-L34
+# changes more rarely than version
+{% set soversion = "11.0.0" %}
```

Comment:
As in #66: We should decide if we want to make use of this to reduce the volume of migrations, and how, see #67

For the record, I'd also be fine to defer any changes around how we treat ABI exports and just go forward with this.